src/gptscript.ts

```diff
@@ -1103,3 +1182,25 @@ function jsonToCredential(cred: string): Credential {
         refreshToken: c.refreshToken
     }
 }
+
+// Dataset types
+
+export type DatasetElementMeta = {
+    name: string
+    description: string
+}
+
+export type DatasetElement = DatasetElementMeta & {
+    contents: string
+}
+
+export type DatasetMeta = {
+    id: string
+    name: string
+    description: string
+}
+
+export type Dataset = DatasetMeta & {
+    baseDir: string
+    elements: Record<string, DatasetElementMeta>
+}
```

Comment:
Everything else is exported as `interface`. Can we do the same here?

Reply:
Yep. The only downside is that, as far as I am aware, I can't do the `&` thing to avoid repeating fields, but I went ahead and changed it all to interfaces.

Reply:
You can use `extends`, but this is fine.

Reply:
Whoops I forgot about that lol. But yeah I agree, it's not a big deal either way